pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/BrokerRegistryImpl.java

```diff
@@ -122,7 +119,7 @@ public boolean isStarted() {
     public synchronized void register() throws MetadataStoreException {
         if (this.state == State.Started) {
             try {
-                this.brokerLookupDataLock = brokerLookupDataLockManager.acquireLock(keyPath(brokerId), brokerLookupData)
+                brokerLookupDataMetadataCache.put(brokerIdKeyPath, brokerLookupData)
```

Comment:
One detail here is that the lock is an ephemeral entry in the metadata store which should get expired when the session dies or disconnects. when switching from acquireLock to put, it becomes a persistent entry in the metadata store. That might be a problem.

Reply:
Good point! I will change the `put` API to accept `CreateOption`

Reply:
It's still confusing to me for why did we introduce `MetadataStoreExtended` and in which case `MetadataStore` is not a `MetadataStoreExtended`. Currently `MetadataCache` just maintains a `MetadataStore` internally. Let me read more details first.

Reply:
I added the `CreateOption` parameter now. PTAL again @lhotari 

BTW, I just read the code about `MetadataStoreExtended`. This API seems not designed well. Currently, built-in implementation all implement this interface. If it does not force the 3rd party library to implement the extra interfaces, it should be documented that where are these methods be used and what will happen if they are not implemented.